Allows version lookup for "foo" to check "foo-core"

toddmowen opened the pull request:
For example, there is no `humbug` artifact, but if the updaterConfig file specifies:

    au.com.cba.omnia % humbug % latest

and the target project's sbt config contains:

    val humbugVersion = xxx

then act as if `humbug` was an artifact name, and fetch the latest version of `humbug-core`.

stephanh commented on src/main/scala/wrangler/api/Automator.scala:
```diff
@@ -30,7 +30,7 @@ case class UpdaterParseError(in: String)
   val msg =
-    s"Couldn't find latest version for ${latest.group} % ${latest.name} in" ++
+    s"Couldn't find latest version for ${latest.group} % ${latest.name} in:\n" ++
     s"${versions.map(_.pretty).mkString("\n")}"
```
I can't believe I wrote this :smile:. Put this inside a string seems completely redundant.

stephanh commented on src/main/scala/wrangler/api/Automator.scala:
```diff
@@ -126,7 +126,11 @@ object Automator {
-      artifacts.find(a => a.group == g && a.name == n)
+      artifacts.collect {
+        case a if a.group == g && a.name == n           => a
+        case a if a.group == g && a.name == n + "-core" => a.copy(name = n)  // match "foo-core" and rewrite artifact to "foo"
```
Why not make it more generic and just say `a.name.startsWith("${n}-")`?

stephanh replied:
What happens if there is a `foo-core` and `foo` artifact?

toddmowen replied:
If there is both `foo-core` and `foo` (on artifactory), then it's undefined which one is used. The assumption is that both would have the same version string anyway.

But a better approach would be to try `foo` first and only fall back to `foo-core` if it is missing. And in this case I would feel comfortable using the `a.name.startsWith("${n}-")` too.

stephanh replied:
You could change the name later.

1. Collect all matches first
2. Try picking matches from the list (exact name first, the prefix match, sorting the list might be enough for this).
3. Change the name to ensure it is the one you want.

stephanh commented:
Can you bump the version number as well please.

toddmowen commented:
Addressed PR comments.

(I originally omitted the version bump because I have multiple PRs open and did not know which one would be accepted first).

stephanh commented:
:+1:

toddmowen commented:
Rebased.
